_rules/element-lang-programmatically-determinable-7ed469.md

```diff
@@ -0,0 +1,142 @@
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+In practice, this rule will be satisfied if the separate success criterion 3.1.1 Language of Page (Level: A) is satisfied. This rule ensures that if 3.1.1 is *not* satisfied, content is checked for alternative ways in which language of page content itself can still have a programmatically determinable human language.
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Language of the `p` element is inherited by the `lang` attribute on the `html` root element.
+
+```html
+<html lang="en">
+	<body>
+		<p>Content</p>
+	</body>
+</html>
+```
+
+#### Passed Example 2
+
+Language of the `p` element is inherited by the `lang` attribute on the `div` parent element. Note that this example fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<div lang="en">
+			<p>Content</p>
+		</div>
+	</body>
+</html>
+```
+
+#### Passed Example 3
+
+Language of the `p` element is inherited by the `lang` attribute on the `div` ancestor element. Note that this example fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<div lang="en">
+			<div>
+				<p>Content</p>
+			</div>
+		</div>
+	</body>
+</html>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Language of the `p` element cannot be programmatically determined as the element itself and none of its ancestors, including the `html` root element, has a defined `lang` or `xml:lang` attribute. Note that this example also fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<p>Content</p>
+	</body>
+</html>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+While the language of the `p` element cannot be programmatically determined, as the element itself and none of its ancestors has a defined `lang` or `xml:lang` attribute, the `p` element contains no actual text content. Note that this example also fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<p></p>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 2
+
+While the language of the `p` element cannot be programmatically determined, as the element itself and none of its ancestors has a defined `lang` or `xml:lang` attribute, the `p` element is not part of the accessibilty tree. Note that this example also fails 3.1.1 Language of Page (Level: A) as the `html` root element lacks a `lang` or `xml:lang` attribute.
+
+```html
+<html>
+	<body>
+		<p aria-hidden="true">Content</p>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 2
```

Comment:
This doesn't add much to the previous Inapplicable Example and can probably be removed.

Reply:
assuming the various ways in which something is hidden/omitted from the accessibility tree gets checked properly, sure

Reply:
Yes. But this rule is not about checking all the possible ways to remove something from the accessibility tree. The examples should illustrate different aspects of the rule, but not the various ways to achieve these aspects.

We could similarly argue that the examples are currently not showing a case where the `lang` attribute is set on the 10th ancestor, or something similar. But we probably agree that we are not going to illustrate all the ways an element can have an ancestor…